collector/collector.go

```diff
@@ -173,7 +195,20 @@ func (c *Collector) collectAllQueues(httpClient *http.Client, result *Result) er
 
 		req = req.WithContext(httptrace.WithClientTrace(req.Context(), trace))
 		if dump, err := httputil.DumpRequest(req, true); err == nil {
-			log.Printf("DEBUG request uri=%s\n%s\n", req.URL, dump)
+			redactedRequestDump := make([]string, 0)
+
+			// Redact Authorization header
+			for _, elem := range strings.Split(string(dump[:]), "\r\n") {
+				if strings.Join(strings.Fields(elem), "") != "" {
+					if strings.HasPrefix(elem, "Authorization") {
+						redactedRequestDump = append(redactedRequestDump, "Authorization: <Redacted>")
+					} else {
+						redactedRequestDump = append(redactedRequestDump, elem)
+					}
+				}
+			}
```

Comment:
Surely there's a more semantic way? e.g. make a shallow copy of the request and its header, and then override the header if it is present: 

```go
dumpReq := *req
if req.Header.Get("Authorization") != "" {
	dumpReq.Header = maps.Clone(req.Header)
	dumpReq.Header.Set("Authorization", "<redacted>")
}
dump, err := httputil.DumpRequest(&dumpReq, true)
if err != nil {
	log.Printf("Couldn't dump request: %v", err)
}
log.Printf("DEBUG request uri=%s\n%s\n", req.URL, dump)

```